addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -622,6 +622,15 @@ var FieldDateRange = InputField.extend({
         }
         this._super.apply(this, arguments);
     },
+    /**
+     * return the UTC date
+     *
+     * @private
+     */
+    _getValue: function () {
+        const value = moment.utc(this.$input.val());
+        return value && value.add(-this.getSession().getTZOffset(value), 'minutes');
```

Comment:
Did you consider using fieldUtils.parse.datetime ? Also, you must be careful: this widget can be set on a date field. In this case, you probably don't want to apply the TZ offset.
So you could do 
```js
return fieldUtils.parse[this.formatType](this.$input.val(), this.field, { timezone: true });
```
unless you have a reason not to.